mlflow/tracking/_tracking_service/client.py

```diff
@@ -80,7 +81,33 @@ def get_metric_history(self, run_id, key):
 
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
         """
-        return self.store.get_metric_history(run_id=run_id, metric_key=key)
+
+        # NB: Paginated query support is currently only available for the RestStore backend.
+        # FileStore and SQLAlchemy store do not provide support for paginated queries and will
+        # raise an MlflowException if the `page_token` argument is not None when calling this
+        # API for a continuation query.
+
+        history = self.store.get_metric_history(
+            run_id=run_id,
+            metric_key=key,
+            max_results=GET_METRIC_HISTORY_MAX_RESULTS,
+            page_token=None,
+        )
+        if not isinstance(history, PagedList):
+            # Backwards compatible return if the server doesn't support paginated queries
+            return history
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/7415/files#r1039930646, I think we can remove this.